cpp/core/shuffle/LocalPartitionWriter.cc

```diff
@@ -387,7 +387,9 @@ std::string LocalPartitionWriter::nextSpilledFileDir() {
 
 arrow::Result<std::shared_ptr<arrow::io::OutputStream>> LocalPartitionWriter::openFile(const std::string& file) {
   std::shared_ptr<arrow::io::FileOutputStream> fout;
-  ARROW_ASSIGN_OR_RAISE(fout, arrow::io::FileOutputStream::Open(file));
+  auto fd = open(file.c_str(), O_WRONLY | O_CREAT | O_TRUNC, S_IRUSR | S_IWUSR | S_IRGRP | S_IROTH);
+  fchmod(fd, 0644);
+  ARROW_ASSIGN_OR_RAISE(fout, arrow::io::FileOutputStream::Open(fd));
```

Comment:
Using `org.apache.spark.shuffle.sort.ColumnarShuffleManager` lacks read permissions for others. The first line is the shuffle data generated by `org.apache.spark.shuffle.sort.ColumnarShuffleManager`, and the second line is the shuffle data generated by Spark built in sort.
<img width="1633" alt="Image" src="https://github.com/user-attachments/assets/aa01e765-c260-4614-8dcb-176f5ffaafe0" />